torch/_inductor/config.py

```diff
@@ -168,6 +168,11 @@ def is_fbcode():
     "raise_comms",
 ]
 
+# enable collective communication function fusion pass (Experimental)
+# e.g., fuse allreduce + split into a single reducescatter)
+# (if pattern_matcher is False, the enable_comm_fusion_pass flag will be ignored)
+enable_comm_fusion_pass = False
```

Comment:
We'll probably add more fusions. Are we going to want to turn them on independently of each other? Make this config accept a list of fusions to run, or just keep it book but name it for allreduce_split fusion. @yf225 what do you think

Reply:
I like the idea of what @yf225 did for reordering. The reordering optimization take a list of string and apply the passes in the list. Users can also add customized pass by adding a function pointer to the list.

Reply:
Since it is unclear how we would group and apply the future distributed optimization passes, I suggest changing the flag naming atm because the current naming is too general as @wconstab also mentioned. ,  @yf225

Reply:
+1 on making the flag name more specific.

In addition I think making it a list could be a good idea because it enforces an order with which we apply the passes (e.g. if user set `A_flag=True` and `B_flag=True`, it's not clear in which order they should be applied; but if user set `passes=[A, B]`, it's more clear that A should be applied before B).

For composability between passes, maybe we can test some common combinations and rely on user reporting issues on their models for edge cases?